amlb/utils/process.py

```diff
@@ -83,6 +84,75 @@ def as_cmd_args(*args, **kwargs):
                        ))
 
 
+def live_output_windows(process: subprocess.Popen, timeout, **ignored) -> Tuple[str, str]:
+    """ Custom output forwarder, because select.select is not Windows compatible. """
+    outputs = dict(
+        out=dict(
+            stream=process.stdout,
+            queue=queue.Queue(),
+            lines=[],
+        ),
+        err=dict(
+            stream=process.stderr,
+            queue=queue.Queue(),
+            lines=[],
+        ),
+    )
+
+    def forward_output(stream, queue_):
+        for line in stream:
+            queue_.put(line)
+
+    for output in outputs.values():
+        t = threading.Thread(target=forward_output, args=(output["stream"], output["queue"]))
+        t.daemon = True
+        t.start()
+
+    while process.poll() is None:
+        for output in outputs.values():
+            while True:
+                try:
+                    line = output["queue"].get(timeout=0.5)
+                    output["lines"].append(line)
+                    print(line.rstrip())
+                except queue.Empty:
+                    break
+    return ''.join(outputs["out"]["lines"]), ''.join(outputs["err"]["lines"])
+
+
+def live_output_unix(process, timeout, input=None, mode='line', **ignored):
```

Comment:
please keep `activity_timeout` for this parameter, and default to `None`.
`timeout` would mean that a call to this function would time out after `timeout`: this is not the case.
However, each time we're trying to read a pipe, if one read times out, then the entire function ends.

Reply:
also you dropped the `mode` logic:
```python
        if mode is True:
            mode = 'line'
```